src/shared/containers/GetFlags.tsx

```diff
@@ -36,7 +37,7 @@ const GetFlags: FC<Props> = ({status, flags}) => {
         return prev
       }, {})
     )
-  }, [flags])
+  }, [Object.keys(flags).length]) // eslint-disable-line react-hooks/exhaustive-deps
```

Comment:
doing comparisons on `flags` was triggering this to run a bunch because JS object comparisons

Reply:
would it be better to `useMemo` here?

Reply:
🤷🏽  what would the benefits of one over the other over both be?

Reply:
I think `useMemo` would help if the same number of flags were present in the array, but the contents of the array had changed. In that case `useMemo` would trigger but checking the length would not. However, I'm not sure if that edge case is applicable here

Reply:
possible to remove this? we want this to update the reporting context every time they change the VALUE of the flags too

Reply:
Totally hear that. Does useEffect do checks against objects for strict equality?

Reply:
I was actually just talking with @ShmuelLotman about this yesterday. The deps array for hooks does not do any deep object comparison. So if you're trying to watch for changes against a non-primitive type, you'll need to do something like `useMemo` to prevent an infinite re-render. And that's because the array's memory reference will change all the time even though the values do not.

Reply:
totally understand deep equality checks, but the reference to this _should_ be changing every time the values are updated (and not touched if they're not changed) through some sort of `flags = {...oldflags, ...newflags}` operation. if something is making that reference change all the time, we should alter the point of ingestion to... not... as the relationship is usually few:many between ingestion and consumption points in the app (few things should be setting this value.. many things should be reading this value)

Reply:
kk, just ended up moving it back to `flags` and we can fix the events that retrigger this call some other time